bool input_0
bool input_1
bool output_0
bool output_1
bool output_2
bool output_3